### PR TITLE
Parse UUID from auth header for anonymous users

### PR DESCRIPTION
**Pull request by daltonfury42**
See Related to https://github.com/SimplQ/simplQ-frontend/issues/477 for details.

Users who have not signed in will send to the backend a UUID generated at client side. This is used to identify the user, till he chooses to register.

Frontend PR: https://github.com/SimplQ/simplQ-frontend/pull/554

### CODE REVIEW

**Review by thehamzarocks — APPROVED** [review]
Do we need checks for UUID collisions since they are being generated client-side? I suppose at our current scale the possibility is extremely unlikely and even if it does happen, it won’t be a significant event

**Comment by daltonfury42** [comment]
> Do we need checks for UUID collisions since they are being generated client-side? I suppose at our current scale the possibility is extremely unlikely and even if it does happen, it won’t be a significant event

Hmm.. unless there is a bug, uuids should not collide. 🤔 

I don't think we will be able to detect collisions easily. 🤔  We would have to add some special header or flag when a UUID is generated, and check for uniqueness at backend side. 🤔 🤔  Or hope that such a bug will never happen. If it does, anonymous users would start managing each other's queue. 

What do you think?

**Comment by thehamzarocks** [comment]
Collisions depend on the nature of randomness and the version of the uuid
generator. But I agree that since detecting and handling such collisions
will be troublesome we don’t need to put in effort there.

Besides, any user who doesn’t sign in needs to accept the risks associated
with anonymous queues.

File: simplq/src/test/java/me/simplq/controller/advices/AuthenticationFilterTest.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import java.net.MalformedURLException;
+import me.simplq.exceptions.SQAccessDeniedException;
+import org.junit.jupiter.api.Test;
+
+class AuthenticationFilterTest {
+  private static final String keyUrl = "https://simplq.us.auth0.com/.well-known/jwks.json";
+
+  @Test
+  void denyBadHeaderValues() throws MalformedURLException {
+    var authFilter = new AuthenticationFilter(new LoggedInUserInfo(), keyUrl);
```
Comment (chalx) [inline simplq/src/test/java/me/simplq/controller/advices/AuthenticationFilterTest.java:16]: A unit test should test only one unit of work. Here you test 6. So we can use `@ParameterizedTest`annotation and the method will look like this:
```java
   @ParameterizedTest
   @ValueSource(strings = {null, "", "invalid header"}) //and so on
   void denyBadHeaderValues(String headerValue) throws MalformedURLException { 
        var authFilter = new AuthenticationFilter(new LoggedInUserInfo(), keyUrl);
       assertThrows(SQAccessDeniedException.class, () -> authFilter.authenticate(headerValue));
  }
```

**Pull request merged**